derive_builder_core/src/build_method.rs

```diff
@@ -67,11 +67,7 @@ impl<'a> ToTokens for BuildMethod<'a> {
         let target_ty = &self.target_ty;
         let target_ty_generics = &self.target_ty_generics;
         let initializers = &self.initializers;
-        let self_param = match self.pattern {
-            BuilderPattern::Owned => quote!(self),
-            BuilderPattern::Mutable |
-            BuilderPattern::Immutable => quote!(&self),
-        };
+        let self_param = self.pattern.to_build_method_tokens();
```

Comment:
I'd like to keep this match statement in `BuildMethod` for the moment. I feel like from the perspective of unit testing this gives clearer boundaries of responsibility. Of course this makes the `BuildMethod` type a little more complex..